swaybar/main.c

```diff
@@ -401,6 +462,247 @@ void render() {
 	}
 }
 
+void free_status_block(void *item) {
+	if (!item) {
+		return;
+	}
+	struct status_block *sb = (struct status_block*)item;
+	if (sb->full_text) { 
+		free(sb->full_text);
+	}
+	if (sb->short_text) {
+		free(sb->short_text);
+	}
+	if (sb->align) {
+		free(sb->align);
+	}
+	if (sb->name) {
+		free(sb->name);
+	}
+	if (sb->instance) {
+		free(sb->instance);
+	}
+	free(sb);
+}
+
+void parse_json(const char *text) {
+/* the array of objects looks like this:
+ * [ {
+ *     "full_text": "E: 10.0.0.1 (1000 Mbit/s)",
+ *     "short_text": "10.0.0.1",
+ *     "color": "#00ff00",
+ *     "min_width": 300,
+ *     "align": "right",
+ *     "urgent": false,
+ *     "name": "ethernet",
+ *     "instance": "eth0",
+ *     "separator": true,
+ *     "separator_block_width": 9
+ *   },
+ *   { ... }, ...
+ * ]
+ */
```

Comment:
Not really. I don't use it, and swaybar ignores align and urgent from the original.